highfidelity/hifi
# Avatar Pipeline Joint Rotation Offsets Handling

This pr is the first step in the avatar pipeline to get non-conforming avatars into High Fidelity.
The case and the spec are located here: https://highfidelity.manuscript.com/f/cases/19421/Implement-Skeleton-Reorienting-Feature-Animation

**Briefly**, we can now read in joint offsets from the FST file for an avatar. The format of the offsets in the FST is the following: jointRotationOffset = Hips = (0.706505, -0.706497, 0.0292632, -0.0292625) where the last four numbers are a quaternion x y z w order.
The rotation represents the global rotation difference between the avatar joint orientation and the high fidelity reference skeleton tpose orientation for that joint.
The absolute rotation is multiplied by this offset and the inverse of the  of the offset is multiplied by the inverseBindPose  in the animSkeleton constructor.  This has the effect of converting animations and incoming tracking rotations into rotations that have the intended effect in the non-standard joint space.

### Test Plan 
1. open interface and walk around in desk top mode and in hmd mode
2. expected result: nothing should have changed for your standard avatars.
3. load the following test avatar.  http://hifi-content.s3.amazonaws.com/angus/avatars/engineer_hifinames/engineer_hifinames_offsets14.fst This avatar has many non-standard joints.    
4. expected result. the avatar should move in desktop and hmd modes like a normal avatar.
